### ISSUE TITLE
AppRemote connect() never finishes

### ISSUE THREAD

**Issue by soapskyway** [spotify/ios-sdk#18]
Trying to set up the examples. Authentication works, the app switches over to Spotify and shows success, starts music and returns to my app. SessionManager->didInitiate is called which does self.appRemote.connect(), the delegate is setup and everything.

The log only shows:

AppRemote: Connecting...

Nothing more, non of the other callbacks is ever called. Is there a way to get more logging to figure out what is going on?

**Comment by jackfreeman** [comment]
@soapskyway If you can provide the following information we can attempt to reproduce the issue.

Spotify App Version:
Sample App You Are Using (and whether or not you've modified it):
iOS Version:
iOS Device: 
XCode Version:


Additionally if you force quit both apps and try again does the problem persist? Have you tried both with and without the xcode debugger connected?

**Comment by soapskyway** [comment]
Hi,

I did the Quick Start steps from https://developer.spotify.com/documentation/ios/quick-start/

Spotify App Version: 8.4.70.686
iOS Version: 11.4.1
IOS Device: IPhone X
Xcode Version: 9.4.1

I think I narrowed it down a bit. The Heroku app can take a long time to spin up so the first time. Maybe some sort of timeout.

If the call to sessionManager.application() returns true, I expect either sessionManager:didInitiateSession or sessionManager:didFailWithError to be called at some point. That doesn't seem to be true. Or maybe the time out is so long I fail to wait for it.

Also seeing a lot of

```
connect
2018-09-16 11:32:03.520437+0200 Sekretariat[709:231494] AppRemote: Connecting...
2018-09-16 11:32:03.525909+0200 Sekretariat[709:231494] AppRemote: Established a connection to the Spotify app.
2018-09-16 11:32:03.622492+0200 Sekretariat[709:231494] AppRemote: Disconnected with error: Error Domain=com.spotify.app-remote.transport Code=-2001 "End of stream." UserInfo={NSLocalizedRecoverySuggestion=Reconnect to the Spotify app., NSLocalizedDescription=End of stream., NSLocalizedFailureReason=One of the streams has reached the end.}
callback
initiate
2018-09-16 11:32:13.756248+0200 Sekretariat[709:231848] AppRemote: Connecting...
2018-09-16 11:32:13.759168+0200 Sekretariat[709:231494] AppRemote: Established a connection to the Spotify app.
2018-09-16 11:32:13.881335+0200 Sekretariat[709:231494] AppRemote: Authenticated with the Spotify app.
2018-09-16 11:32:13.881469+0200 Sekretariat[709:231494] AppRemote: Established a session with the Spotify app.
appRemote connected
```

I kind of like the new API, and I understand it is the first version, but things feel a bit shaky. Wish you could share a bit of the internal implementation of the API. Or expose some more ways to log or detect what is going on.

Anyway.. thanks a lot for the API and help using it :)

**Comment by soapskyway** [comment]
Okay, the problem with the double connections comes from how the Quick start instructions are done. Both the session setup flow and the applicationDidBecomeActive(_ application: UIApplication) cause a connection to be performed. https://developer.spotify.com/documentation/ios/quick-start/

**Comment by ethan021021** [comment]
I was running into this issue as well. Please update your docs Spotify!

**Comment by 19neloyk** [comment]
If you are still running into this, I think I found a solution (one that works for me at the very least). 

MY ISSUE: (SKIP THIS PARAGRAPH IF YOU DO NOT CARE ABOUT CONTEXT)
In my case, I had to send a connection request from the main thread of my application. If you have structured your program similar to how I have, you probably have **appRemote.connect()** inside of **sessionManager(manager:,didInitiate session:)** after we are granted a successful session from the sessionManager.initiateSession().  I think, from my hours of looking at this stupid issue, that code within session is to be executed on a background thread incorrectly, and this might be why it does not get looked at (something to do with the prioritization of other threads). 
Note: I don't exactly know why it works for the sample app in the sdk, but I am sure that it has something to do with initialization of remote probably.

**BEFORE**(WRONG):
`func sessionManager(manager: SPTSessionManager, didInitiate session: SPTSession) {
          appRemote.connectionParameters.accessToken = session.accessToken
          appRemote.connect() // This is on a background thread
}
`
**AFTER**(RIGHT):
`func sessionManager(manager: SPTSessionManager, didInitiate session: SPTSession) {
          appRemote.connectionParameters.accessToken = session.accessToken
          DispatchQueue.main.async {       //Everything in this block is now on the main thread
                 appRemote.connect()                
          }
}
`
